Tests/MongoSwiftTests/Document+SequenceTests.swift

```diff
@@ -0,0 +1,192 @@
+        return [
+            ("testIterator", testIterator),
+            ("testMapFilter", testMapFilter),
+            ("testDropFirst", testDropFirst),
+            ("testDropLast", testDropLast),
+            ("testDropPredicate", testDropPredicate),
+            ("testPrefixLength", testPrefixLength),
+            ("testPrefixPredicate", testPrefixPredicate),
+            ("testSuffix", testSuffix),
+            ("testSplit", testSplit)
+        ]
+    }
+
+    func testIterator() {
+        let doc: Document = [
+            "string": "test string",
+            "true": true,
+            "false": false,
+            "int": 25,
+            "int32": Int32(5),
+            "double": Double(15),
+            "decimal128": Decimal128("1.2E+10"),
+            "minkey": MinKey(),
+            "maxkey": MaxKey(),
+            "date": Date(timeIntervalSince1970: 5000),
+            "timestamp": Timestamp(timestamp: 5, inc: 10)
+        ]
+
+        for (_, _) in doc { }
+    }
+
+    func testMapFilter() throws {
+        let doc1: Document = ["a": 1, "b": nil, "c": 3, "d": 4, "e": nil]
+        expect(doc1.mapValues { $0 ?? 1 }).to(equal(["a": 1, "b": 1, "c": 3, "d": 4, "e": 1]))
+        let output1 = doc1.mapValues { val in
+            if let int = val as? Int { return int + 1 }
+            return val
+        }
+        expect(output1).to(equal(["a": 2, "b": nil, "c": 4, "d": 5, "e": nil]))
+        expect(doc1.filter { $0.value != nil }).to(equal(["a": 1, "c": 3, "d": 4]))
+
+        let doc2: Document = ["a": 1, "b": "hello", "c": [1, 2] as [Int]]
+        expect(doc2.filter { $0.value is String }).to(equal(["b": "hello"]))
+        let output2 = doc2.mapValues { val in
+            switch val {
+            case let val as Int:
+                return val + 1
+            case let val as String:
+                return val + " there"
+            case let val as [Int]:
+                return val.reduce(0, +)
+            default:
+                return val
+            }
+        }
+        expect(output2).to(equal(["a": 2, "b": "hello there", "c": 3]))
+    }
+
+    // shared docs for subsequence tests
+    let emptyDoc = Document()
+    let smallDoc: Document = ["x": 1]
+    let doc: Document = ["a": 1, "b": "hi", "c": [1, 2] as [Int], "d": false, "e": nil, "f": MinKey(), "g": 10]
+
+    // shared predicates for subsequence tests
+    func isInt(_ pair: Document.KeyValuePair) -> Bool { return pair.value is Int }
+    func isNotNil(_ pair: Document.KeyValuePair) -> Bool { return pair.value != nil }
+    func is10(_ pair: Document.KeyValuePair) -> Bool {
+        if let int = pair.value as? Int { return int == 10 } else { return false }
+    }
+    func isNot10(_ pair: Document.KeyValuePair) -> Bool { return !is10(pair) }
+
+    func testDropFirst() throws {
+        expect(self.emptyDoc.dropFirst(0)).to(equal([:]))
+        expect(self.emptyDoc.dropFirst(1)).to(equal([:]))
```

Comment:
I assume this behavior (dropping from an empty sequence returns the empty sequence) is based on Dictionary's `dropFirst()`?